zerver/lib/narrow.py

```diff
@@ -1276,6 +1297,25 @@ def fetch_messages(
     )
 
     with get_sqlalchemy_connection() as sa_conn:
+        if anchor_date:
+            # The anchor_date value needs to be parsed here because the date query can
+            # only be applied after all the other narrow conditions have been applied.
```

Comment:
Why can it only be applied at the end?

Reply:
I've improved the comment to clarify this:
```.py
# `date_anchor` is not a filter but a mechanism to obtain the anchor value.
# This anchor is utilized to restrict the query within a specific range.
# It can only be computed at this point, considering the set of messages/query
# that have already undergone narrow conditions.
```